includes/payments/class-payments-query.php

```diff
@@ -609,12 +618,22 @@ private function remap_args() {
 			$arguments['date_created_query']['inclusive'] = true;
 		}
 
+		if ( isset( $this->initial_args['number'] ) ) {
+			if ( -1 == $this->initial_args['number'] ) {
```

Comment:
Is it safe to do a strict check here?

Reply:
You could technically pass in the number as a string and if we do a strict check here then it wouldn't work the way we want (to fetch all results).

Both of these should work the same:

```php
$query = new EDD_Payments_Query( array(
	'number' => -1
) );
$query->get_payments();
```

```php
$query = new EDD_Payments_Query( array(
	'number' => '-1'
) );
$query->get_payments();
```

Reply:
What about forcing `$this->initial_args['number']` to a string -- that seems to be the most common?

Reply:
If we're going to force it to something I think I'd rather do an integer. That's what all our default number values are already.

Is there something in particular you're concerned about here with not using a strict check though?

Reply:
No big deal since we don't technically enforce it, but ideally it will eventually become a rule: https://github.com/WordPress/WordPress-Coding-Standards/blob/develop/WordPress/Sniffs/PHP/StrictComparisonsSniff.php